Makefile

```diff
@@ -0,0 +1,95 @@
+# Image URL to use all building/pushing image targets
+IMG ?= controller:latest
+# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
+CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+all: build
+
+verify: fmt vet lint
+
+# Run tests
+test: generate verify manifests unit
+
+unit:
+	hack/unit-tests.sh
+
+# Build operator binaries
+build: operator
+
+operator:
+	go build -o bin/cluster-controller-manager-operator cmd/cluster-cluster-api-operator/main.go
+
+# Run against the configured Kubernetes cluster in ~/.kube/config
+run: verify manifests
+	go run cmd/cluster-cluster-api-operator/main.go
+
+# Generate manifests e.g. CRD, RBAC etc.
+manifests: controller-gen
+	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+
+# Run go fmt against code
+.PHONY: fmt
+fmt:
+	go fmt ./...
+
+# Run go vet against code
+.PHONY: vet
+vet:
+	go vet ./...
+
+# Run golangci-lint against code
+.PHONY: lint
+lint: golangci-lint
+	( GOLANGCI_LINT_CACHE=$(PROJECT_DIR)/.cache $(GOLANGCI_LINT) run --timeout 10m )
+
+# Run go mod
+.PHONY: vendor
+vendor:
+	go mod tidy
+	go mod vendor
+	go mod verify
+
+# Generate code
+generate: controller-gen
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+
+# Build the docker image
+.PHONY: image
+image:
+	docker build -t ${IMG} .
+
+# Push the docker image
+.PHONY: push
+push:
+	docker push ${IMG}
```

Comment:
Not everyone use docker. Can we wrap it?

Aslo, there is no dockerfile . Does this targets make sense?

Reply:
I will add a Dockerfile in a follow-up and yes I can default to podman as in other MAPI components.

Reply:
Then, i would suggest to remove targets which is not needed / does not work now and add it by necessity.